azurerm/resource_arm_application_gateway.go

```diff
@@ -2282,6 +2306,37 @@ func flattenApplicationGatewayRedirectConfigurations(input *[]network.Applicatio
 	return results, nil
 }
 
+func expandAutoscaleConfiguration(d *schema.ResourceData) *network.ApplicationGatewayAutoscaleConfiguration {
+	vs := d.Get("autoscale_configuration").([]interface{})
+	v := vs[0].(map[string]interface{})
```

Comment:
This is panicking when `autoscale_configuration` hasn't been set. We can do something like the below to kick us out of this function if `autoscale_configuration` hasn't been set. 
```suggestion
      if len(vs) == 0 {
        return nil
      }
      v := vs[0].(map[string]interface{})
```